src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5346,6 +5343,13 @@ private bool IsTrueIdentifier()
             return false;
         }
 
+        private bool IsTrueIdentifier(SyntaxToken token)
```

Comment:
Good idea, but not quite so simple to do, as there are a series of helper methods for the previous method intended to help it disambiguate, implicitly based on the current token in declarations; it peeks forward using `PeekToken(int)` and even does some trial forward parsing.

This method is for use in executable code (not declarations), so it has a much simpler implementation that depends only on the token passed in and `this.IsInQuery`. I'll add some clarifying doc comments.